File: js/11057.js
const fs = require('fs');

const N = +fs.readFileSync('example.txt').toString();

const dp = Array(1001)
  .fill()
  .map(() => Array(10).fill(0));

for (let i = 0; i < 10; i++) {
  dp[1][i] = 1;
}

for (let i = 2; i <= N; i++) {
  dp[i][0] = 1;
  for (let j = 1; j < 10; j++) {
    dp[i][j] = (dp[i][j - 1] + dp[i - 1][j]) % 10007;
  }
}

const result = dp[N].reduce((a, b) => a + b, 0);
console.log(result % 10007);

File: c++/11057.cpp
#include <iostream>

using namespace std;

// dp[i][j] : 길이가 i이고 마지막번째자리가 j인 오르막 수의 개수
int dp[1001][10];

int main() {
  freopen("example.txt", "r", stdin);
  int N;
  cin >> N;

  for (int i = 0; i < 10; i++) {
    dp[1][i] = 1;
  }

  for (int i = 2; i <= N; i++) {
    dp[i][0] = 1;
    for (int j = 1; j < 10; j++) {
      dp[i][j] = (dp[i][j - 1] + dp[i - 1][j]) % 10007;
    }
  }

  int result = 0;
  for (int i = 0; i < 10; i++) {
    result += dp[N][i];
  }
  cout << result % 10007;

  return 0;
}